### ISSUE TITLE
`--verify` is buggy sometimes

### ISSUE THREAD

**Issue by zhiqiangxu** [foundry-rs/foundry#5006]
### Component

Forge

### Have you ensured that all of these are up to date?

- [ ] Foundry
- [ ] Foundryup

### What version of Foundry are you on?

forge 0.2.0 (e2fa2b5 2023-02-19T00:15:20.51416Z)

### What command(s) is the bug in?

forge script --verify

### Operating System

macOS (Intel)

### Describe the bug

It repeatedly reports:

```
Contract verification status:
Response: `NOTOK`
Details: `Pending in queue`
```

and finally reports:

```
Error: 
Checking verification result failed:
```

But after manually checking the verification result at the explorer side, I found the contracts in fact have already been verified.

**Comment by daltyboy11** [comment]
I think this is a problem with your etherscan API key @zhiqiangxu. You might be getting rate limited: https://info.etherscan.com/api-return-errors/

<img width="999" alt="Screenshot 2023-05-23 at 4 41 04 PM" src="https://github.com/foundry-rs/foundry/assets/19829228/f173635e-29ec-4964-964c-e185812841a6">

**Comment by Evalir** [comment]
Agreed with above. If you're using blockscout, we'll be also be shipping improvements soon that should solve this. Apart from this, verification APIs are not great, so not much we can do other than try to build stable wrappers around them.

**Comment by zhiqiangxu** [comment]
So in case the api is rate limited, maybe foundry should output the exact error message and lower down the calling frequency?

**Comment by zhiqiangxu** [comment]
> I think this is a problem with your etherscan API key @zhiqiangxu. You might be getting rate limited: https://info.etherscan.com/api-return-errors/
> 
> <img alt="Screenshot 2023-05-23 at 4 41 04 PM" width="999" src="https://user-images.githubusercontent.com/19829228/240414053-f173635e-29ec-4964-964c-e185812841a6.png">

So in case the api is rate limited, maybe foundry should output the exact error message and lower down the calling frequency?

**Comment by Evalir** [comment]
Closing as verification was improved recently on ethers—and sadly we can't do much on the API side.
